app/assets/src/components/common/CoverageVizBottomSidebar/HitGroupViz.jsx

```diff
@@ -0,0 +1,293 @@
+    }
+  };
+
+  handleContigCopy = async () => {
+    const { contigDownloaderData } = this.state;
+
+    const data = await getContigsSequencesByByteranges(
+      this.props.sampleId,
+      contigDownloaderData.contigByteranges,
+      this.props.pipelineVersion
+    );
+
+    copy(Object.values(data).join("\n"));
+
+    this.setState({
+      currentCopyIconMessage: "Copied to clipboard!"
+    });
+  };
+
+  restoreCopyIconMessage = () => {
+    this.setState({
+      currentCopyIconMessage: DEFAULT_CONTIG_COPY_MESSAGE
+    });
+  };
+
+  renderGenomeVizTooltip = () => {
+    const { genomeVizTooltipLocation, genomeVizTooltipData } = this.state;
+
+    return ReactDOM.createPortal(
+      <div
+        style={getTooltipStyle(genomeVizTooltipLocation)}
+        className={cs.hoverTooltip}
+      >
+        <DataTooltip data={genomeVizTooltipData} />
+      </div>,
+      document.body
+    );
+  };
+
+  renderContigDownloader = () => {
+    const { contigDownloaderLocation } = this.state;
+
+    return ReactDOM.createPortal(
+      <div
+        style={{
+          left: contigDownloaderLocation.left,
+          top: contigDownloaderLocation.top - 5
+        }}
+        className={cs.contigDownloader}
+        ref={c => (this._contigDownloader = c)}
+      >
+        <BasicPopup
+          trigger={
+            <div
+              className={cx(cs.icon, cs.downloadIcon)}
+              onClick={this.handleContigDownload}
+            >
+              <DownloadIcon />
+            </div>
+          }
+          inverted
+          wide="very"
+          content="Download Contig FASTA"
+          onClick={this.handleContigDownload}
+        />
+        <BasicPopup
+          trigger={
+            <div
+              className={cx(cs.icon, cs.copyIcon)}
+              onClick={this.handleContigCopy}
+              onMouseEnter={this.restoreCopyIconMessage}
+            >
+              <CopyIcon />
+            </div>
+          }
+          inverted
+          wide="very"
+          content={this.state.currentCopyIconMessage}
+        />
+      </div>,
+      document.body
+    );
+  };
+
+  render() {
+    const {
+      genomeVizTooltipLocation,
+      genomeVizTooltipData,
+      contigDownloaderLocation,
+      contigDownloaderData
+    } = this.state;
+
+    return (
+      <div className={cs.genomeVizRow}>
+        <div className={cs.rowLabel}>Contigs and Reads</div>
+        <div
+          className={cs.genomeViz}
+          ref={hitGroupVizContainer => {
+            this.hitGroupVizContainer = hitGroupVizContainer;
+          }}
+        />
+        {genomeVizTooltipLocation &&
+          genomeVizTooltipData &&
+          this.renderGenomeVizTooltip()}
+        {contigDownloaderLocation &&
+          contigDownloaderData &&
+          this.renderContigDownloader()}
+      </div>
+    );
+  }
+}
+
+HitGroupViz.propTypes = {
+  accessionData: PropTypes.shape({
+    avg_prop_mismatch: PropTypes.number,
```

Comment:
why were these ones snake_case again? because they come from the Python files?

Reply:
They come directly from a JSON file where it's snake_case. :(